// element-plus/lib/components/table-v2/src/table.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index.js');
var common = require('./common.js');
var row = require('./row.js');
var header = require('./header.js');
var grid = require('./grid.js');
var props = require('../../../utils/vue/props.js');

const tableV2Props = props.buildProps({
  cache: grid.tableV2GridProps.cache,
  estimatedRowHeight: row.tableV2RowProps.estimatedRowHeight,
  rowKey: common.rowKey,
  cellProps: {
    type: props.definePropType([Object, Function])
  },
  headerClass: {
    type: props.definePropType([
      String,
      Function
    ])
  },
  headerProps: {
    type: props.definePropType([
      Object,
      Function
    ])
  },
  headerCellProps: {
    type: props.definePropType([
      Object,
      Function
    ])
  },
  headerHeight: header.tableV2HeaderProps.headerHeight,
  footerHeight: {
    type: Number,
    default: 0
  },
  rowClass: {
    type: props.definePropType([String, Function])
  },
  rowProps: {
    type: props.definePropType([Object, Function])
  },
  rowHeight: Number,
  columns: common.columns,
  data: common.dataType,
  fixedData: common.fixedDataType,
  expandColumnKey: row.tableV2RowProps.expandColumnKey,
  expandedRowKeys: common.expandKeys,
  defaultExpandedRowKeys: common.expandKeys,
  class: common.classType,
  disabled: Boolean,
  fixed: Boolean,
  style: {
    type: props.definePropType([String, Array, Object])
  },
  width: common.requiredNumber,
  height: Number,
  maxHeight: Number,
  sortBy: {
    type: props.definePropType(Object),
    default: () => ({})
  },
  onColumnSort: {
    type: props.definePropType(Function)
  },
  onExpandedRowsChange: Function,
  onRowExpand: Function,
  onScroll: grid.tableV2GridProps.onScroll,
  rowEventHandlers: row.tableV2RowProps.rowEventHandlers
});

exports.tableV2Props = tableV2Props;
