// element-plus/es/components/table-v2/src/grid.mjs
import '../../../utils/index.mjs';
import '../../virtual-list/index.mjs';
import { columns, dataType, fixedDataType, requiredNumber, classType, styleType } from './common.mjs';
import { tableV2HeaderProps } from './header.mjs';
import { tableV2RowProps } from './row.mjs';
import { buildProps, definePropType } from '../../../utils/vue/props.mjs';
import { virtualizedListProps } from '../../virtual-list/src/props.mjs';

const tableV2GridProps = buildProps({
  columns,
  data: dataType,
  fixedData: fixedDataType,
  estimatedRowHeight: tableV2RowProps.estimatedRowHeight,
  width: requiredNumber,
  height: requiredNumber,
  headerWidth: requiredNumber,
  headerHeight: tableV2HeaderProps.headerHeight,
  bodyWidth: requiredNumber,
  rowHeight: requiredNumber,
  cache: virtualizedListProps.cache,
  rowKey: tableV2RowProps.rowKey,
  useIsScrolling: Boolean,
  class: classType,
  style: styleType,
  containerStyle: styleType,
  onRowRendered: {
    type: Function
  },
  onScroll: {
    type: definePropType(Function)
  }
});

export { tableV2GridProps };
